### PR TITLE
Return a 500 for _ping if there is no elected primary

### PR DESCRIPTION
**Pull request by wsong**

### CODE REVIEW

File: api/replica.go [hunk 1]
```diff
@@ -50,3 +54,12 @@ func (p *Replica) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		httpError(w, fmt.Sprintf("Unable to reach primary cluster manager (%s): %v", err, p.primary), http.StatusInternalServerError)
 	}
 }
+
+func (p *Replica) ping(w http.ResponseWriter, r *http.Request) {
+	if p.primary == "" {
+		httpError(w, "No elected primary cluster manager", http.StatusInternalServerError)
+		w.Write([]byte{'E', 'R', 'R', 'O', 'R'})
```
Comment (wsong) [inline api/replica.go]: Heh. I wanted to maintain consistency with this:

https://github.com/docker/swarm/blob/master/api/handlers.go#L963

But I'll go ahead and change it to your version.

**Comment by nishanttotla** [comment]
LGTM

Ping @dongluochen

**Comment by dongluochen** [comment]
Essentially you are trying to change `/_ping` from local request to a cluster request. I think it can be done like below. 

```
 - var localRoutes = []string{"/info", "/_ping", "/debug"}
 + var localRoutes = []string{"/info", "/debug"}
```

**Comment by wsong** [comment]
If I understand this correctly, we don't want to just blindly forward _ping requests because the primary might be down, in which case we want _ping to return a 500.

**Comment by dongluochen** [comment]
If you remove `/_ping` from the `localRoutes`, it would come to this block, right? 

```
 	// Otherwise, forward.
 	if p.primary == "" {
 		httpError(w, "No elected primary cluster manager", http.StatusInternalServerError)
 		return
 	}
```

**Comment by wsong** [comment]
Oh yeah, right. Done.

**Comment by dongluochen** [comment]
LGTM

**Pull request merged**